web/controllers/userController/index.ts

```diff
@@ -1007,7 +1011,66 @@ async function putActive(wix_id: string, active: boolean, person: Pupil | Studen
                 await dissolveMatch(matches[i], 0, person);
             }
 
-            // Step 2: Deactivate
+            // Step 2: Cancel all courses if user is student
+            if (type == Student) {
+                let courses = await getConnection()
+                    .getRepository(Course)
+                    .createQueryBuilder("course")
+                    .leftJoinAndSelect("course.instructors", "instructors")
+                    .getMany();
+
+                courses.forEach(async (course: Course) => {
```

Comment:
This _fires and forgets_ all transactions, which I guess is not intended ... I'd rather do a `for(const course of courses)` for sequential processing or a `await Promise.all(courses.map(...))` for proper error handling ....

Reply:
Oh yeah, I wasn't aware of that. Thanks